interfaces/apparmor/spec.go

```diff
@@ -220,20 +223,20 @@ func isProbablyPresent(path string) bool {
 	return path == "/" || path == "/snap" || path == "/var" || path == "/var/snap" || path == "/tmp" || path == "/usr" || path == "/etc"
 }
 
-// WritableMimicProfile writes apparmor rules for a writable mimic at the given path.
-func WritableMimicProfile(buf *bytes.Buffer, path string, assumedPrefixDepth int) {
-	fmt.Fprintf(buf, "  # Writable mimic %s\n", path)
+// GenWritableMimicProfile generates apparmor rules for a writable mimic at the given path.
+func GenWritableMimicProfile(emit func(f string, args ...interface{}), path string, assumedPrefixDepth int) {
```

Comment:
You can use something like:
```
type emitFunc func(f string, args ...interface{})

func GenWritableMimicProfile(emit emitFunc, path string, assumedPrefixDepth int) {
   ...
}
```
Or use whatever name seems fitting given the other PRs.

Reply:
I'm happy to iterate on this after the bug fix is merged.